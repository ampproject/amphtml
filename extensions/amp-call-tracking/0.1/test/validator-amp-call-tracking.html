<!--
  Test Description:
  Tests for the amp-call-tracking tag.
-->
<!doctype html>
<html âš¡ lang="en">
<head>
  <meta charset="utf-8">
  <link rel="canonical" href="./regular-html-version.html">
  <meta name="viewport" content="width=device-width,minimum-scale=1">
  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
  <script async custom-element="amp-call-tracking" src="https://cdn.ampproject.org/v0/amp-call-tracking-latest.js"></script>
  <script async src="https://cdn.ampproject.org/v0.js"></script>
</head>
<body>
  <!-- Valid: Example of amp-call-tracking usage -->
  <amp-call-tracking config="https://example.com/calltracking.json">
    <a href="tel:12345678">+1 (23) 45-678</a>
  </amp-call-tracking>

  <!-- Invalid: no <A> tag -->
  <amp-call-tracking config="https://example.com/calltracking.json">
  </amp-call-tracking>

  <!-- Invalid: more than one <A> tag -->
  <amp-call-tracking config="https://example.com/calltracking.json">
    <a href="tel:12345678">+1 (23) 45-678</a>
    <a href="tel:98765432">(987) 654-321</a>
  </amp-call-tracking>

  <!-- Invalid: no "config" attribute -->
  <amp-call-tracking>
    <a href="tel:12345678">+1 (23) 45-678</a>
  </amp-call-tracking>

  <!-- Invalid: Something other than an <A> child -->
  <amp-call-tracking config="https://example.com/calltracking.json">
    <div></div>
  </amp-call-tracking>
</body>
</html>
