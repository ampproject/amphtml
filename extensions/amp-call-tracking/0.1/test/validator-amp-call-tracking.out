FAIL
|  <!--
|    Copyright 2017 The AMP HTML Authors. All Rights Reserved.
|
|    Licensed under the Apache License, Version 2.0 (the "License");
|    you may not use this file except in compliance with the License.
|    You may obtain a copy of the License at
|
|        http://www.apache.org/licenses/LICENSE-2.0
|
|    Unless required by applicable law or agreed to in writing, software
|    distributed under the License is distributed on an "AS-IS" BASIS,
|    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
|    See the License for the specific language governing permissions and
|    limitations under the license.
|  -->
|  <!--
|    Test Description:
|    Tests for the amp-call-tracking tag.
|  -->
|  <!doctype html>
|  <html âš¡>
|  <head>
|    <meta charset="utf-8">
|    <link rel="canonical" href="./regular-html-version.html">
|    <meta name="viewport" content="width=device-width,minimum-scale=1">
|    <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
|    <script async custom-element="amp-call-tracking" src="https://cdn.ampproject.org/v0/amp-call-tracking-latest.js"></script>
|    <script async src="https://cdn.ampproject.org/v0.js"></script>
|  </head>
|  <body>
|    <!-- Valid: Example of amp-call-tracking usage -->
|    <amp-call-tracking config="https://example.com/calltracking.json">
|      <a href="tel:12345678">+1 (23) 45-678</a>
|    </amp-call-tracking>
|
|    <!-- Invalid: no <A> tag -->
|    <amp-call-tracking config="https://example.com/calltracking.json">
>>   ^~~~~~~~~
amp-call-tracking/0.1/test/validator-amp-call-tracking.html:37:2 Tag 'amp-call-tracking' must have 1 child tags - saw 0 child tags. (see https://amp.dev/documentation/components/amp-call-tracking/)
|    </amp-call-tracking>
|
|    <!-- Invalid: more than one <A> tag -->
|    <amp-call-tracking config="https://example.com/calltracking.json">
>>   ^~~~~~~~~
amp-call-tracking/0.1/test/validator-amp-call-tracking.html:41:2 Tag 'amp-call-tracking' must have 1 child tags - saw 2 child tags. (see https://amp.dev/documentation/components/amp-call-tracking/)
|      <a href="tel:12345678">+1 (23) 45-678</a>
|      <a href="tel:98765432">(987) 654-321</a>
|    </amp-call-tracking>
|
|    <!-- Invalid: no "config" attribute -->
|    <amp-call-tracking>
>>   ^~~~~~~~~
amp-call-tracking/0.1/test/validator-amp-call-tracking.html:47:2 The mandatory attribute 'config' is missing in tag 'amp-call-tracking'. (see https://amp.dev/documentation/components/amp-call-tracking/)
|      <a href="tel:12345678">+1 (23) 45-678</a>
|    </amp-call-tracking>
|
|    <!-- Invalid: Something other than an <A> child -->
|    <amp-call-tracking config="https://example.com/calltracking.json">
|      <div></div>
>>     ^~~~~~~~~
amp-call-tracking/0.1/test/validator-amp-call-tracking.html:53:4 Tag 'div' is disallowed as child of tag 'amp-call-tracking'. Child tag must be one of ['a']. (see https://amp.dev/documentation/components/amp-call-tracking/)
|    </amp-call-tracking>
|  </body>
|  </html>
